core/src/main/scala/org/apache/spark/ui/SparkUI.scala

```diff
@@ -167,7 +167,7 @@ private[spark] class SparkUI private (
         attemptId = None,
         startTime = new Date(startTime),
         endTime = new Date(-1),
-        duration = 0,
+        duration = System.currentTimeMillis() - startTime,
```

Comment:
This wont work when used in context of history server

Reply:
This is SparkUI only, not used by HistoryServer

Reply:
Seems OK, I just wonder why it was ever always 0?

Reply:
https://github.com/apache/spark/pull/10648/files#diff-32f7bbe4f1ae694bd35f180c2ff918ed7ce5da42f37560bbd4d5ade6c7c2783eR116

It was initially merged with a giant PR, which was for HistoryServer. So I guess this line didn't get much attention

Reply:
`SparkUI` is used from history server - not sure if this specific api is used though.

Reply:
> `SparkUI` is used from history server

Only the method `create` from the companion object of `SparkUI` is referenced by history server. The classes `SparkUI` and `HistoryServer` both inherit from `WebUI`, they don't share functionality with each other.

Reply:
Sounds good, thanks for confirming !